R/clustering.R

```diff
@@ -82,6 +85,13 @@ FindClusters.default <- function(
             edge.file.name = edge.file.name
           )
         } else if (algorithm == 4) {
+          if (method == "matrix" || method == "igraph") {
+            if (method == "igraph") {
+              object <- graph_from_adjacency_matrix(adjmatrix = object)
+            }
+          } else {
+            stop("method must be 'matrix' or 'igraph'")
+          }
```

Comment:
Why is this check happening if we're parallelizing the clustering but not otherwise? Also, shouldn't this check be in `RunLeiden` to avoid duplicating code?

Reply:
Ah, didn't realise there was a separate RunLeiden call in the parallel version. Yes let's move it there so it's called by both.